### ISSUE TITLE
Understanding BILOU tagging in vw

### ISSUE THREAD

**Issue by dvbarnes** [VowpalWabbit/vowpal_wabbit#2163]
I'm at a loss currently as to why my simple BILOU tagger is not working using VW and was hoping I could get some insight into why.

I'm trying to predict a tag for the following train.txt and test.txt files
**train.txt**

> 1 | the
> 7 | United
> 8 | States
> 8 | of
> 9 | America
> 1 | agreements
> 1 | is
> 1 | hereby
> 1 | voted
> 1 | as
> 1 | established
> 
> 1 | the
> 7 | United
> 8 | States
> 8 | of
> 9 | America
> 1 | is
> 1 | great
> 
> 1 | the
> 7 | United
> 8 | States
> 8 | of
> 9 | America
> 1 | is
> 1 | happy

**test.txt**

> 1 | the
> 7 | United
> 8 | States
> 8 | of
> 9 | America
> 1 | is
> 1 | big
> 
> 1 | the
> 7 | United
> 8 | States
> 8 | of
> 9 | America
> 1 | is
> 1 | good
> 

according to https://github.com/VowpalWabbit/vowpal_wabbit/blob/f032eaf50086b6814ae1169e633620b2b05bfd47/vowpalwabbit/search_sequencetask.cc#L77 my assumption is 
7 -> B-Country
8 -> I-Country
9 -> L-Country

now when I run my training:
which I found https://github.com/VowpalWabbit/vowpal_wabbit/blob/f032eaf50086b6814ae1169e633620b2b05bfd47/test/RunTests#L1066
```
vw -k -c -d train.txt --passes 20 --search_alpha 1e-8 --search_task sequencespan \
    --search_span_bilou --search 9 --holdout_off -f data.model
```

I get the output:
> final_regressor = data.model
> switching to BILOU encoding for sequence span labeling
> Num weight bits = 18
> learning rate = 0.5
> initial_t = 0
> power_t = 0.5
> decay_learning_rate = 1
> creating cache_file = train.txt.cache
> Reading datafile = train.txt
> num sources = 1
> average    since      instance            current true      current predicted   cur   cur   predic    cache  examples
> loss       last        counter           output prefix          output prefix  pass   pol     made     hits    gener  beta
> 2.000000   2.000000          1  [1 1 8 8 9 1 1 1 1 ..] [1 1 1 1 1 1 1 1 1 ..]     0     0       11        0       11  0.000000
> 2.000000   2.000000          2  [1 1 8 8 9 1 1       ] [1 1 1 1 1 1 1       ]     0     0       18        0       18  0.000000
> 1.000000   0.000000          4  [1 1 8 8 9 1 1       ] [1 1 8 8 9 1 1       ]     0     0       32        0       32  0.000000
> 0.500000   0.000000          8  [1 1 8 8 9 1 1       ] [1 1 8 8 9 1 1       ]     1     0       71        0       71  0.000001
> 0.250000   0.000000         16  [1 1 8 8 9 1 1       ] [1 1 8 8 9 1 1       ]     3     0      149        0      149  0.000001
> 0.125000   0.000000         32  [1 1 8 8 9 1 1       ] [1 1 8 8 9 1 1       ]     7     0      305        0      305  0.000003
> 0.062500   0.000000         64  [1 1 8 8 9 1 1       ] [1 1 8 8 9 1 1       ]    15     0      617        0      617  0.000006
> 
> finished run
> number of examples per pass = 4
> passes used = 20
> weighted example sum = 81.000000
> weighted label sum = 0.000000
> average loss = 0.049383
> total feature number = 2340

and the testing command which I took from https://github.com/VowpalWabbit/vowpal_wabbit/blob/f032eaf50086b6814ae1169e633620b2b05bfd47/test/RunTests#L1073

`vw -d test.txt -t --search_span_bilou -i data.model -p pred.txt`

my _pred.txt_ file seems to be missing my B tag and I can't for the life of me figure out why.

> 1 1 8 8 9 1 1
> 1 1 8 8 9 1 1

My assumptions are that since the sentences are very similar there shouldn't be much learning that needs to take place. And that I'm just trying to get a basic example up and running to start from.

**Comment by dvbarnes** [comment]
Moved to [stack overflow](https://stackoverflow.com/questions/59141596/understanding-bilou-tagging-in-vowpal-wabbit) as it's more appropriate there
